screens/Directions.js

```diff
@@ -0,0 +1,305 @@
+    const [isFirstInstruction, setIsFirstInstruction] = React.useState(false);
+    const mapRef = useRef(null);
+ 
+    /* 2. Read the params from the navigation state */
+    const { params } = props.navigation.state;
+    const destinationResponse = params ? params.destinationResponse : null;
+
+    useEffect(() => {
+        if (instructionIndex  == 0) {
+            setIsFirstInstruction(true); //When Direction enabled, we are at the first instruction.
+        }
+      });
+    
+      /**
+       * Description: Go back method
+       * Using Stack navigator.
+       */
+    const goBackPressHandler = () => {
+        props.navigation.goBack();
+    }
+    /**
+     * Description: This method will update the region to fit all coordinates of the a specific (index) step of the instruction. 
+     * @param {*} index 
+     */
+    const updateMapRegionToInstruction = (index) => {
+        setTimeout(() => {
+            mapRef.current.fitToCoordinates(
+                destinationResponse.steps[index].polylines,
+                { edgePadding: { bottom: 100, right: 50, left: 50, top: 300 }, animated: true, });
+            
+        }, 100);
+    }
+
+    /**
+     * Description: The map has to be initialized to the first instruction.
+     * Particularity: We used the reference of the mapview used in ref={mapRef}. 
+     */
+    const initMapRegion = () => {
+        setTimeout(() => {
+            mapRef.current.fitToCoordinates(
+                destinationResponse.steps[0].polylines,
+                { edgePadding: { bottom: 100, right: 50, left: 50, top: 300 }, animated: true, });
+        }, 100);
+    }
+    /**
+     * Description: This method will increment the instructionIndex so that we can render the next instruction 
+     * in the MapView. 
+     * How: 
+     * 1. Making sure it's not the last instruction, if it is, we disabled the button by setting 
+     * setIsLastInstruction to true. If it's not, we increment the counter and setIsLastInstruction
+     * to false.
+     * 2. We update the mapView by calling updateMapRegionToInstruction with the index incremented.
+     */
+    const goToNextInstruction = () => {
+        if(instructionIndex >= destinationResponse.steps.length - 1) {
+            setIsLastInstruction(true);
+        } else {
+        setInstructionIndex(instructionIndex + 1);
+        setIsFirstInstruction(false);
+        updateMapRegionToInstruction(instructionIndex + 1);      
+        }
+    }
+    /**
+     * Same as goToNextInstruction but for previous instruction
+     */
+    const goToPreviousInstruction = () => {
+        if(instructionIndex == 0) {
+            setIsFirstInstruction(true);
+        } else {
+        setInstructionIndex(instructionIndex - 1);
+        setIsLastInstruction(false);
+        updateMapRegionToInstruction(instructionIndex - 1);
+        }
+    }
+
+    return (
+        <View>
+            <MapView
```

Comment:
With a new MapView, how are we mixing both indoor & outdoor?

Reply:
The new MapView is more closer to the home Mapview than you think. I just didnt add the components (buildingHighlights and etc). A way to mix indoor and outdoor would be to implement the Indoor functionalities into Directions and every time a classroom is inputted in the search context, we extend the detailed instructions to have start point (indoor View) or/and last instruction to be (indoor View).

Reply:
Baton, needs investigation for sprint 4